### PR TITLE
Lock protection for replica data structures during REST API handling

### PR DESCRIPTION
**Pull request by vishnuitta**
In replicas of jiva volumes, during REST API handling, replica related data structures are used without any lock protection.
This leads to data race between reader and writer of those structures.

This PR is to lock protect of those structures during REST API handling.

This PR also adds logs to track REST APIs and its errors.

### CODE REVIEW

File: replica/server.go [hunk 1]
```diff
@@ -188,14 +189,32 @@ func (s *Server) PrevStatus() (State, Info) {
...
 	}
+	return stats1
 }
 
 func (s *Server) GetUsage() (*types.VolUsage, error) {
-	return s.r.GetUsage()
+	if s.r != nil {
+		return s.r.GetUsage()
+	}
+	return &types.VolUsage{
```
Comment (pawanpraka1) [inline replica/server.go]: why this change? shouldn't we return error in this case?

Comment (vishnuitta) [reply replica/server.go]: error seems to be ignored in the code, which is why just returning error may not be sufficient. It needs nil handling in the callers.
Added a TODO for nil handling
